lua/nvim_lsp/configs.lua

```diff
@@ -110,10 +110,10 @@ function configs.__newindex(t, config_name, config_def)
 
       add_callbacks(new_config)
       if config_def.on_new_config then
-        pcall(config_def.on_new_config, new_config)
+        pcall(config_def.on_new_config, new_config, _root_dir)
```

Comment:
This change makes it possible to use the newly resolved root directory in any config transformations that are needed and doesn't cause any breaking changes.